### ISSUE TITLE
getSymbolsWithAnnotation does not handle type aliases

### ISSUE THREAD

**Issue by yigit** [google/ksp#506]
For the given code:

```
annotation class Anno
typealias AnnoAlias = Anno

@AnnoAlias
class UsesAlias

```

`resolver.getSymbolsWithAnnotation("Anno")` does not return `UsesAlias`.

I wasn't initially sure if this is a bug or not but given that in this case, the user is creating the alias, Symbol processor has no knowledge of it so I think KSP should handle this (for reference, works fine in KAPT but that is expected since it turns into java code).

**Comment by yigit** [comment]
repro case:
https://github.com/yigit/ksp/commit/d876ef11c857722e53ccab428a59faced1de05ba

**Comment by ting-yuan** [comment]
Fixed in #580
